### PR TITLE
Save full response to a file

### PR DESCRIPTION
**Pull request by aldreth**
* Add a new button on the response preview pane
* Save full response to file when button clicked

closes https://github.com/getinsomnia/insomnia/issues/192

### CODE REVIEW

**Comment by gschier** [comment]
@aldreth I'll take a closer look at this tomorrow but it looks great after a quick glance. 

One thing, since this changes UI, can you attach a screenshot of the new dropdown?

**Comment by aldreth** [comment]
NP 😄 
screen shot 2017-05-14 at 22 15 56

**Comment by aldreth** [comment]
@gschier I couldn't see any tests you'd written for saving the body or anything, so didn't try to add any for this.

File: app/ui/components/dropdowns/preview-mode-dropdown.js [hunk 1]
```diff
@@ -35,6 +35,10 @@ class PreviewModeDropdown extends PureComponent {
           <i className="fa fa-save"/>
           Save to File
```
Comment (gschier) [inline app/ui/components/dropdowns/preview-mode-dropdown.js]: Should we rename this one to `"Save Request Body"`?

File: app/ui/components/response-pane.js [hunk 2]
```diff
@@ -81,6 +81,47 @@ class ResponsePane extends PureComponent {
...
+      title: 'Save Full Response',
+      buttonLabel: 'Save',
+      filters: [{
+        name: 'Download', extensions: [extension]
+      }]
+    };
+
+    remote.dialog.showSaveDialog(options, filename => {
+      if (!filename) {
+        trackEvent('Full Response', 'Save Cancel');
```
Comment (gschier) [inline app/ui/components/response-pane.js]: These `trackEvent` calls should keep the first parameter (category) as `Response`. The "Full" descriptor is better placed in the second parameter (action). 

Eg. `trackEvent('Response', 'Save Full Cancel');`

File: app/ui/components/dropdowns/preview-mode-dropdown.js (hunk 1 above)
Comment (aldreth) [inline app/ui/components/dropdowns/preview-mode-dropdown.js]: Makes sense.

File: app/ui/components/response-pane.js (hunk 2 above)
Comment (aldreth) [inline app/ui/components/response-pane.js]: NP

**Comment by aldreth** [comment]
@gschier Updated code to reflect your comments. I also spotted that I'd labelled the dropdown as `Save Full Request`, so I've fixed that, and I've removed the dependency of the file extension on the content type, as it doesn't really apply when saving full request.

**Comment by gschier** [comment]
Thanks @stewarnet for the extra set of 👀

**Comment by aldreth** [comment]
Cheers for the help with this @gschier & for the 👀 @stewarnet.

**Pull request merged**